### PR TITLE
Parallel batch processing

### PR DESCRIPTION
**Pull request by Max-K-cadence**
Description:
This merge request introduces a series of patches that allow Image Classifier to handle
minibatches in parallel when running in -minibatch mode.

Testing: Few models in consecutive and parallel moves.

### CODE REVIEW

File: tools/loader/ImageClassifier.cpp [hunk 1]
```diff
@@ -412,80 +444,130 @@ int main(int argc, char **argv) {
...
-  std::vector<std::string> inputImageBatchFilenames;
-  if ((!miniBatchMode) && (!streamInputFilenamesMode)) {
-    inputImageBatchFilenames = inputImageFilenames;
-  }
-
   // Print out the inferred image classification.
-  llvm::outs() << "Model: " << loader.getFunction()->getName() << "\n";
-
+  llvm::outs() << "Model: " << Loader::getModelOptPath() << "\n";
+  std::mutex io_mu;
```
Comment (jackm321) [inline tools/loader/ImageClassifier.cpp]: camel case

**Comment by Max-K-cadence** [comment]
Rebased & addressed comments.

**Comment by Max-K-cadence** [comment]
Friendly ping.

File: tools/loader/ImageClassifier.cpp [hunk 2]
```diff
@@ -453,10 +479,9 @@ setupContextPool(Placeholder *outputPH, Placeholder *inputImagePH,
 }
 
 int main(int argc, char **argv) {
-  PlaceholderBindings bindings;
-  // The loader verifies/initializes command line parameters, and initializes
+  // Verify/initialize command line parameters, and then loader initializes
```
Comment (jackm321) [inline tools/loader/ImageClassifier.cpp]: period

**Pull request closed without merging**